components/service/glean/src/main/java/mozilla/components/service/glean/storages/StringListsStorageEngine.kt

```diff
@@ -0,0 +1,82 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.service.glean.storages
+
+import android.annotation.SuppressLint
+import mozilla.components.service.glean.CommonMetricData
+import mozilla.components.support.base.log.logger.Logger
+
+/**
+ * This singleton handles the in-memory storage logic for strings. It is meant to be used by
+ * the Specific Strings API and the ping assembling objects. No validation on the stored data
+ * is performed at this point: validation must be performed by the Specific Strings API.
+ *
+ * This class contains a reference to the Android application Context. While the IDE warns
+ * us that this could leak, the application context lives as long as the application and this
+ * object. For this reason, we should be safe to suppress the IDE warning.
+ */
+@SuppressLint("StaticFieldLeak")
+internal object StringListsStorageEngine : StringListsStorageEngineImplementation()
+
+internal open class StringListsStorageEngineImplementation(
+    override val logger: Logger = Logger("glean/StringsStorageEngine")
+) : GenericScalarStorageEngine<Array<String>?>() {
```

Comment:
Curiosity: why is this an `Array` and not a `List`? Is this intentional (I'm not saying this is wrong).

Reply:
It should be a List<String>, you are correct.  I had originally used an array as I had an issue with List but I went back and looked at it again and I can get it to work with List.